solarforecastarbiter/validation/tasks.py

```diff
@@ -37,6 +50,58 @@ def _solpos_night(observation, values):
     return solar_position, night_flag
 
 
+def _resample_date_range(interval_length, closed, freq, values):
+    # consider moving this to utils
+    data_start, data_end = values.index[0], values.index[-1]
```

Comment:
I don't know if it's possible that this will get an emtpy frame/series, but should probably double check upstream and handle if that is possible or comment that it isn't

Reply:
`tasks.apply_validation` prevents this in production. The relevant test is `test_apply_validation_empty`. I suspect there are other problems in the tasks/validators with empty data. However, `_solpos_night_instantaneous` does actually work with empty data:

```
>>> tasks._solpos_night_instantaneous(obs, pd.DataFrame())

(Empty DataFrame
 Columns: [apparent_zenith, zenith, apparent_elevation, elevation, azimuth, equation_of_time]
 Index: [], Series([], Name: zenith, dtype: int64))
```

I could add a statement to `_solpos_night_resample` along the lines of `if values.empty: return _solpos_night_instantaneous(observation, values)`. 

I'll leave it `tasks.apply_validation` unless you feel otherwise.